daffodil-lib/src/main/scala/org/apache/daffodil/util/Logger.scala

```diff
@@ -17,213 +17,30 @@
 
 package org.apache.daffodil.util
 
-import java.text.SimpleDateFormat
-import java.util.Date
-import java.io.File
-import java.io.PrintStream
-import java.io.FileOutputStream
-import org.apache.daffodil.exceptions.Assert
-import org.apache.daffodil.util._
-import org.apache.daffodil.util.Maybe._
-import org.apache.daffodil.exceptions.UnsuppressableException
+import org.apache.logging.log4j.scala.Logging
 
 /**
- * Simple logging system evolved from code found on Stack Overflow, on the web.
- * http://stackoverflow.com/questions/2018528/logging-in-scala
- * Mostly based on the contribution of Don Mackenzie.
+ * The log4j.scala.Logging trait adds a log4j.scala.Logger member val called
+ * 'logger' to whatever class mixes it in. Classes that mixin this trait can
+ * then just call logger.warn/info/debug/etc to log a message, which are macros
+ * to minimize overhead.
  *
- * Extensively modified to use Macros for performance now.
+ * However, the Logger class is not serializable, and so all classes that might
+ * need to be serialized, (e.g. runtime objects), cannot take this approach.
+ *
+ * Instead, we have a single Logger object that mixes in this trait. This
+ * object is never serialized and so can be used anywhere, regardless of
+ * serializablility. For simplicity, consistency, and to avoid potential
+ * serialization issues, all logging should be done via the logger in this
+ * object, rather than mixing in the Logging trait, for example:
+ *
+ *   Logger.log.info("Message to log")
+ *
+ * The downside to this is that it breaks the ability to use Log4j's feature to
+ * configure different log levels for different namespaces or classes or have
+ * useful line numbers, because all logging comes from and is associated with
+ * this single Logger in the 'org.apache.daffodil.util.Logger' class.
```

Comment:
We could have multiple logger objects if we wanted to, so as to get some of these capabilities back. Not sure we need it, but it seems possible.

Reply:
Yep, I considered this and it should work, but I figured that could be a future enhancement if we determine it's needed.